src/core/jit/builder.rs

```diff
@@ -360,12 +362,26 @@ impl Builder {
             is_introspection_query,
         );
 
+        // perform the rule check.
+        Rules
+            .pipe(
+                QueryComplexity::new(self.index.query_complexity().unwrap_or_default())
+                    .when(self.index.query_complexity().is_some()),
+            )
+            .pipe(
+                QueryDepth::new(self.index.query_depth().unwrap_or_default())
+                    .when(self.index.query_depth().is_some()),
+            )
+            .validate(&plan)
+            .to_result()?;
```

Comment:
Store the computed query depth in the plan. The builder should not fail if depth or complexity thresholds are exceeded.